plugins/pulp_rpm/plugins/distributors/yum/publish.py

```diff
@@ -579,109 +547,109 @@ def finalize(self):
                                            self.context.checksum)
 
 
-class PublishRpmAndDrpmStepIncremental(UnitPublishStep):
+class PublishRpmAndDrpmStepIncremental(platform_steps.UnitModelPluginStep):
     """
-    Publish all incremental errata
+    Publish all incremental rpms and drpms
     """
     def __init__(self, **kwargs):
         super(PublishRpmAndDrpmStepIncremental, self).__init__(constants.PUBLISH_RPMS_STEP,
-                                                               [TYPE_ID_RPM, TYPE_ID_SRPM,
-                                                                TYPE_ID_DRPM],
-                                                               unit_fields=PACKAGE_FIELDS, **kwargs)
+                                                               [models.RPM, models.SRPM,
+                                                                models.DRPM], **kwargs)
         self.description = _('Publishing RPM, SRPM, and DRPM')
 
-    def process_unit(self, unit):
+    def initialize(self):
+        """
+        In case there are no units that get processed, nothing else would create this directory.
+        Its existence is required by the CopyDirectoryStep.
+        """
+        super(PublishRpmAndDrpmStepIncremental, self).initialize()
+        if not os.path.exists(self.get_working_dir()):
+            os.makedirs(self.get_working_dir())
+
+    @property
+    def unit_querysets(self):
+        """
+        Limits the queryset's fields
+
+        :return:    generator of mongoengine.QuerySet objects that have fields limited
+        :rtype:     generator
+        """
+        querysets = super(PublishRpmAndDrpmStepIncremental, self).unit_querysets
+        # The repodata field can be huge, and we don't need it right now.
+        return (qs.exclude('repodata') for qs in querysets)
+
+    def process_main(self, item=None):
         """
         Link the unit to the content directory and the package_dir
 
         :param unit: The unit to process
-        :type unit: pulp.plugins.model.Unit
+        :type unit: pulp.server.db.model.ContentUnit
         """
-        source_path = unit.storage_path
-        relative_path = util.get_relpath_from_unit(unit)
+        unit = item
+        source_path = unit._storage_path
+        relative_path = unit.filename
         destination_path = os.path.join(self.get_working_dir(), relative_path)
-        self._create_symlink(source_path, destination_path)
+        plugin_misc.create_symlink(source_path, destination_path)
 
         filename = unit.unit_key['name'] + '-' + unit.unit_key['version'] + '-' + \
```

Comment:
Either way will work. The direct attribute access as ``.name`` is probably preferable.